packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -46,17 +58,28 @@ export class B3Propagator implements HttpTextPropagator {
   inject(context: Context, carrier: unknown, setter: SetterFunction) {
     const spanContext = getParentSpanContext(context);
     if (!spanContext) return;
-
+    const parentSpanId = context.getValue(PARENT_SPAN_ID_KEY);
     if (
       isValidTraceId(spanContext.traceId) &&
       isValidSpanId(spanContext.spanId)
     ) {
+      if (parentSpanId) {
+        if (isValidTraceId(parentSpanId as string)) {
```

Comment:
I believe this needs to be `isValidSpanId` instead of `isValidTraceId`.

Reply:
oh, yeah. I thought I had changed that. Thanks.